### PR TITLE
Add tests for discovery ACL functionality

### PR DESCRIPTION
**Pull request by bpgray**
This PR secures the consul test instance with a default ACL policy of 'deny', and then runs a functional service registration test with a given ACL.

For cleanliness, the secured consul instance should allow registrations for the other tests without an ACL. Other tests do currently pass, so this isn't necessarily required given the current test design.

### CODE REVIEW

**Comment by bpgray** [comment]
Suspect build is failing due to #83
The build worked on my fork in travis after commenting out the ./docs/src/main/asciidoc/ghpages.sh command

File: src/main/bash/local_run_consul.sh [hunk 1]
```diff
@@ -1,2 +1,5 @@
 #!/bin/bash
-consul agent -server -bootstrap-expect 1 -data-dir /tmp/consul -ui-dir `dirname $0`/../../../src/test/resources/consul_ui
+consul agent -server -bootstrap-expect 1 -data-dir /tmp/consul -ui-dir `dirname $0`/../../../src/test/resources/consul_ui -config-dir `dirname $0`/../../test/resources/consul_config &
```
Comment (spencergibb) [inline src/main/bash/local_run_consul.sh]: Can we create a 2nd bash script, maybe `travis_run_consul.sh`, that has the config for tokens? That way I can run without tokens normally, but for builds use the new script.

**Comment by bpgray** [comment]
Failing due to #84 
A script in the PR is removed by ghpages.sh

File: src/main/bash/local_run_consul.sh (hunk 1 above)
Comment (bpgray) [inline src/main/bash/local_run_consul.sh]: 2nd bash script created and now used in .travis.yml

**Comment by spencergibb** [comment]
I haven't forgot about this. I need to make it so these tests only run on the travis build, not on a local build.

**Comment by bpgray** [comment]
I think the tests will still work on a local build providing you haven't hardened your local consul instance with different acl keys. The test basically makes sure the tokens works against a hardened consul instance.

**Comment by spencergibb** [comment]
Thats the thing, I don't want to run it with acls.

**Comment by spencergibb** [comment]
Squashed, merged with a bit of polish via 12af0f532e1b60c51bf3e6e64fc0daacffa197e6

**Comment by spencergibb** [comment]
@bpgray this is merged, but the config tests fail now, so for now they are disabled.

**Pull request closed without merging**